boards/components/src/button.rs

```diff
@@ -6,7 +6,8 @@
 //! let button = components::button::ButtonComponent::new(board_kernel).finalize(
 //!     components::button_component_helper!((
 //!         &sam4l::gpio::PC[24],
-//!         capsules::button::GpioMode::LowWhenPressed
+//!         capsules::button::GpioMode::LowWhenPressed,
+//!         kernel::hil::gpio::FloatingState::PullUp
```

Comment:
+1 for making board developers think about this, but maybe give some documentation / guidance on how to make the choice? Maybe here or maybe lower in the component declaration?

Reply:
When I modified the code I was actually hesitating adding assert statements as safeguards to ensure that developers can't activate pull-downs when the button is `LowWhenPressed` or pull-up when it's `HighWhenPressed`. Both cases would result in the button being seen as permanently pressed (meh) as well as always drawing current and potentially dissipating extra-heat (bad!).

Reply:
So maybe just a comment in `button.rs` saying that typically `LowWhenPressed` should be paired with `PullUp`? (Or something like that?)